include/texture.hpp

```diff
@@ -7,7 +7,15 @@
 
 class Texture {
 public:
-    Texture(const int width, const int height);
+    enum TextureType{
+        DEPTH=GL_DEPTH_COMPONENT,
+        COLOR=GL_RGB,
+    };
+    enum TextureFormat{
+        CUBE=GL_TEXTURE_CUBE_MAP,
+        TEXTURE=GL_TEXTURE_2D,
+    };
```

Comment:
i think this can be renamed to `enum Format` and `enum Target` respectively according to the [`glTexImage2D` documentation](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glTexImage2D.xhtml)

Reply:
done